pkg/agent/monitortool/monitor.go

```diff
@@ -354,7 +363,40 @@ func (m *NodeLatencyMonitor) pingAll(ipv4Socket, ipv6Socket net.PacketConn) {
 			klog.ErrorS(nil, "Cannot send ICMP message to Node IP because socket is not initialized for IP family", "IP", toIP)
 		}
 	}
-	klog.V(4).InfoS("Done pinging all Nodes")
+	klog.InfoS("Done pinging all Nodes")
+}
+
+// GetSummary returns the latency summary of the given Node IP.
+func (m *NodeLatencyMonitor) GetSummary() *stv1aplpha1.NodeLatencyStats {
+	nodeName, _ := env.GetNodeName()
```

Comment:
this is not great: you should retrieve the Node name once only, when you instantiate `NodeLatencyMonitor`.

Reply:
Note that the name is part of the `NodeConfig` too, so there is no need to check the environment

Reply:
That's right, I updated it from `m.nodeName`.